### PR TITLE
[FIX] account: fix account type computation

### PR DESCRIPTION
**Pull request by aliyatastemirova**
There can be many accounts in self and the `search_read` is currently breaking because of `self.company_id` in the domain.
Account codes should however be still limited to the current company and we don't want to do a `search_read` for account codes within the for loop, so one way is to prefetch and create a nested dictionary.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_account.py [hunk 1]
```diff
@@ -374,14 +375,19 @@ def _compute_opening_debit_credit(self):
...
         """ Compute the account type based on the account code.
         Search for the closest parent account code and sets the account type according to the parent.
+        If there is no parent (e.g. the account code is lower than any other existing account code),
+        the account type will be set to 'asset_current'.
         """
         accounts_to_process = self.filtered(lambda r: r.code and not r.account_type)
-        accounts_with_codes = {r['code']: r['account_type'] for r in self.search_read(domain=[(
-            'code', '!=', False), ('company_id', '=', self.company_id.id)], fields=['code', 'account_type'])}
-        codes_list = list(accounts_with_codes.keys())
+        all_accounts = self.search_read(domain=[('code', '!=', False), ('company_id', 'in', self.company_id.ids)], fields=['code', 'account_type', 'company_id'])
```
Comment (Levizar) [inline addons/account/models/account_account.py:382]: We only need those from companies of `accounts_to_process` which might be a subset
```suggestion
        all_accounts = self.search_read(domain=[('code', '!=', False), ('company_id', 'in', accounts_to_process.company_id.ids)], fields=['code', 'account_type', 'company_id'])
```

**Review by Levizar — CHANGES_REQUESTED** [review]
Only one small code change that might be insignificant otherwise the code LGTM but the commit message could me clearer and straightforward.

You are solving 2 different problems but it's not obvious from the commit message:
- multi-company environment
- Default value for account.type

Put a small sentence below each to explain the problem.

Bonus just to mess around with you: :grin:  
> we don't want to search_read for account codes within
the for loop, so one way is to prefetch and create a nested dictionary

This appear subjective. Instead of it, you can write something like "for performance reason we batch the searching part" which is more factual.

Otherwise, nice job!

**Review by Levizar — APPROVED** [review]

File: addons/account/models/account_account.py [hunk 2]
```diff
@@ -374,14 +375,19 @@ def _compute_opening_debit_credit(self):
...
         """ Compute the account type based on the account code.
         Search for the closest parent account code and sets the account type according to the parent.
+        If there is no parent (e.g. the account code is lower than any other existing account code),
+        the account type will be set to 'asset_current'.
         """
         accounts_to_process = self.filtered(lambda r: r.code and not r.account_type)
-        accounts_with_codes = {r['code']: r['account_type'] for r in self.search_read(domain=[(
-            'code', '!=', False), ('company_id', '=', self.company_id.id)], fields=['code', 'account_type'])}
-        codes_list = list(accounts_with_codes.keys())
+        all_accounts = self.search_read(domain=[('code', '!=', False), ('company_id', 'in', accounts_to_process.company_id.ids)], fields=['code', 'account_type', 'company_id'])
```
[the remark this answers is not in the document]
Comment (aliyatastemirova) [reply addons/account/models/account_account.py]: you're right, I fixed that :slightly_smiling_face:

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**